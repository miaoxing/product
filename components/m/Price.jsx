import {Text} from '@fower/taro';
import PropTypes from 'prop-types';

const Price = ({children}) => {
  const price = children.toString();
  const [integer, decimal] = price.split('.');

  return (
    <Text textLg primary500>
      <Text textXs>ï¿¥</Text>
      {integer}
      {!!decimal && <Text textXs>.{decimal}</Text>}
    </Text>
  );
};

Price.propTypes = {
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.object]),
};

export default Price;
