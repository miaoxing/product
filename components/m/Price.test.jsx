import Price from './Price';
import {render} from '@testing-library/react';

describe('Price', () => {
  test('basic', () => {
    const {container} = render(<Price>1.1</Price>);
    expect(container).toMatchSnapshot();
  });

  test('int', () => {
    const {container} = render(<Price>12</Price>);
    expect(container).toMatchSnapshot();
  });

  test('float', () => {
    const {container} = render(<Price>{12.2}</Price>);
    expect(container).toMatchSnapshot();
  });
});
