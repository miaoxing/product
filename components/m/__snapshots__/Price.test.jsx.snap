// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Price basic 1`] = `
<div>
  <taro-text-core
    primary500="true"
  >
    <taro-text-core>
      ￥
    </taro-text-core>
    1
    <taro-text-core>
      .
      1
    </taro-text-core>
  </taro-text-core>
</div>
`;

exports[`Price float 1`] = `
<div>
  <taro-text-core
    primary500="true"
  >
    <taro-text-core>
      ￥
    </taro-text-core>
    12
    <taro-text-core>
      .
      2
    </taro-text-core>
  </taro-text-core>
</div>
`;

exports[`Price int 1`] = `
<div>
  <taro-text-core
    primary500="true"
  >
    <taro-text-core>
      ￥
    </taro-text-core>
    12
  </taro-text-core>
</div>
`;
